### DISCUSSION TITLE
How do I return a pure data structure instead of key:value like ?

### DISCUSSION THREAD

**Discussion by joisun** [prisma/prisma#16700] [Q&A]
Here is my code for quering a blog post :

```ts
const res = await this.prisma.article.findUnique({
  where: {
    artid: id,
  },
  include: {
    column: {
      select: {
        colid: true,
      },
    },
    tags: {
      select: {
        tagid: true,
      },
    },
    category: {
      select: {
        catid: true,
      },
    },
  },
});
```

And which return value below:

```ts
{
  "artid": "67050f46-67cb-440e-a2af-943db2a2f8e5",
  "sort": 0,
  "authorId": "cf2de9ea-fd01-4c0b-bdad-43da5359fe12",
  "title": "Nestjs-Authentication 文档翻译",
  "description": "这是一个文章摘要",
  "content": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
  "keywords": "",
  "status": 0,
  "cover": "",
  "visible": true,
  "password": "12345",
  "allowComment": true,
  "liking": 0,
  "readers": 0,
  "banner": 1,
  "catid": "b8ba38bb-5585-461f-942d-604cb1ba378d",
  "colid": "edc4d493-6d5d-4a13-948c-f2ec15985607",
  "createdAt": "2022-12-08T16:32:15.939Z",
  "updatedAt": "2022-12-08T16:32:15.939Z",
  "column": {
    "colid": "edc4d493-6d5d-4a13-948c-f2ec15985607"
  },
  "tags": [
    {
      "tagid": "5984049e-9242-40eb-b529-00cde6ca0dd3"
    },
    {
      "tagid": "dabf3f3c-4af2-47be-9ccc-36a5d7fa7655"
    }
  ],
  "category": {
    "catid": "b8ba38bb-5585-461f-942d-604cb1ba378d"
  }
}
```

Things work fine , but I need to convert the fields before I use them in frontend,like :

```js
  "column": {
    "colid": "edc4d493-6d5d-4a13-948c-f2ec15985607"
  }
  // TO:
  "column":"edc4d493-6d5d-4a13-948c-f2ec15985607"
```

```js
  "category": {
    "catid": "b8ba38bb-5585-461f-942d-604cb1ba378d"
  }
  // TO:
  "category": "b8ba38bb-5585-461f-942d-604cb1ba378d"
```

```js
  "tags": [
    {
      "tagid": "5984049e-9242-40eb-b529-00cde6ca0dd3"
    },
    {
      "tagid": "dabf3f3c-4af2-47be-9ccc-36a5d7fa7655"
    }
  ],
  //TO:
  "tags":["5984049e-9242-40eb-b529-00cde6ca0dd3","dabf3f3c-4af2-47be-9ccc-36a5d7fa7655"]
```

so, is there any way to make things easier for me to get what I expect directly ? thanks!
(btw, here is my prisma file if needed [link](https://github.com/jaycethanks/zhongyi-blog-server/blob/master/prisma/schema.prisma))

**Comment by cjdunteman** [comment]
I don't believe you can do this in the query itself because you would be essentially redefining "column" from a relation to a field with a string field type (the column id).

**Reply by pantharshit00** [reply]
Hello @jaycethanks, did you get a chance to try the above? Please let us know if this is still open.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
